dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BundleRestRepository.java

```diff
@@ -140,6 +144,7 @@ public BitstreamRest uploadBitstream(Context context, Bundle bundle, String file
                 itemService.update(context, item);
             }
             bundleService.update(context, bundle);
+            provenanceService.uploadBitstream(context, bundle);
```

Comment:
wouldn't bundleService be a better place?

Reply:
No, because it doesn’t always make sense to add provenance when the update method is called. We’ve had this discussion a couple of times.

Reply:
ok but still, give me a valid reason

Reply:
The bundleService.update method is used in 15 places. I think it is quite risky to add a provenance method there, as we don't know it will work properly or if it might cause duplications,..

Reply:
It cannot be in the bitstreamService.update() because it is called 2 times during the single request here:

PackageUtils.addDepositLicense()
	bitstreamService.create()
		bitstreamStorageServiceImpl.store()
			**bitstreamService.update()**
	**bitstreamService.update()**